pareto/tests/test_strategic_model.py

```diff
@@ -472,6 +482,71 @@ def test_basic_reduced_build_capex_distance_based_capacity_input(
     assert isinstance(m.PipelineExpansionCapEx, pyo.Constraint)
 
 
+@pytest.mark.unit
+def test_basic_reduced_build_discrete_water_quality_input(
+    build_reduced_strategic_model,
+):
+    """Make a model and make sure it doesn't throw exception"""
+    m = build_reduced_strategic_model(
+        config_dict={
+            "objective": Objectives.cost,
+            "pipeline_cost": PipelineCost.capacity_based,
+            "pipeline_capacity": PipelineCapacity.input,
+            "water_quality": WaterQuality.discrete,
+        }
+    )
+    assert degrees_of_freedom(m) == 109453
```

Comment:
I looked at why the tests are failing, it looks like this value needs to be updated to 109457

assert degrees_of_freedom(m) == 109453
E       assert 109457 == 109453

Reply:
adjusted the value